### PR TITLE
Use RTGS.global Id instead of Did

### PR DESCRIPTION
**Pull request by lewishenson**
Non-breaking changes for messages that are sent to/from RTGS - hence duplicate properties or headers. Will allow the changes to be gradually rolled out.

Breaking changes for SDK-specific items, e.g. the options. When new version of the package is published, apps can be simply updated to use the renamed properties.

### CODE REVIEW

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/IdCryptPublisher.cs [hunk 1]
```diff
@@ -16,19 +16,19 @@ public IdCryptPublisher(IInternalPublisher internalPublisher)
 
 	public Task<SendResult> SendIdCryptInvitationToBankAsync(
 		IdCryptInvitationV1 message,
-		string bankPartnerDid,
+		string bankPartnerRtgsGlobalId,
 		CancellationToken cancellationToken)
 	{
-		var headers = new Dictionary<string, string> { { "bankpartnerdid", bankPartnerDid } };
+		var headers = new Dictionary<string, string> { { "bankpartnerdid", bankPartnerRtgsGlobalId }, { "bank-partner-rtgs-global-id", bankPartnerRtgsGlobalId } };
```
Comment (chris-brown) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/IdCryptPublisher.cs:22]: I vote for Pascal Kebab case

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Messages/IdCryptBankInvitationNotificationV1.cs [hunk 2]
```diff
@@ -10,6 +10,11 @@ public record IdCryptBankInvitationNotificationV1
 	/// </summary>
 	public string BankPartnerDid { get; init; }
```
Comment (chris-brown) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Messages/IdCryptBankInvitationNotificationV1.cs:11]: can deprecate this property with xml docs.

Comment (chris-brown) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Messages/IdCryptBankInvitationNotificationV1.cs:11]: helping with removing it

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs [hunk 3]
```diff
@@ -158,7 +158,7 @@ private async Task EnsureRtgsCallSetup(CancellationToken cancellationToken)
 
 		if (_toRtgsCall is null)
 		{
-			var grpcCallHeaders = new Metadata { new("bankdid", _options.BankDid) };
+			var grpcCallHeaders = new Metadata { new("bankdid", _options.RtgsGlobalId), new("rtgs-global-id", _options.RtgsGlobalId) };
 			_toRtgsCall = _paymentClient.ToRtgsMessage(grpcCallHeaders, cancellationToken: CancellationToken.None);
```
Comment (chris-brown) [inline src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/InternalPublisher.cs:162]: Same as headers for RtgsMessage. I vote for Pascal-Kebab case

**Review by chris-brown — CHANGES_REQUESTED** [review]

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/IdCryptPublisher.cs (hunk 1 above)
Comment (lewishenson) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/IdCryptPublisher.cs]: Agreed

File: src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Messages/IdCryptBankInvitationNotificationV1.cs (hunk 2 above)
Comment (lewishenson) [reply src/RTGS.DotNetSDK/RTGS.DotNetSDK/Publisher/IdCrypt/Messages/IdCryptBankInvitationNotificationV1.cs]: No longer applicable as switching to a breaking change approach.

**Review by chris-brown — APPROVED** [review]

**Comment by lewishenson** [comment]
Change of approach, now going with breaking changes.

**Pull request merged**